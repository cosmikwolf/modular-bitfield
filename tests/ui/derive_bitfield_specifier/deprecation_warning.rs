#![deny(deprecated)]

use modular_bitfield::prelude::*;

#[derive(BitfieldSpecifier)]
enum Foo {
    Zero = 0,
    One = 1,
}

#[bitfield]
#[derive(BitfieldSpecifier)]
struct Bar {
    f: u8,
}

#[bitfield]
#[derive(BitfieldSpecifier)]
struct Baz {
    f: u8,
}

fn main() {}
