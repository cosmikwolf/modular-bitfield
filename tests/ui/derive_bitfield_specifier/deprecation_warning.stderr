error: use of deprecated macro `BitfieldSpecifier`: use #[derive(Specifier)]. This alias will be removed in 0.13.
 --> tests/ui/derive_bitfield_specifier/deprecation_warning.rs:5:10
  |
5 | #[derive(BitfieldSpecifier)]
  |          ^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> tests/ui/derive_bitfield_specifier/deprecation_warning.rs:1:9
  |
1 | #![deny(deprecated)]
  |         ^^^^^^^^^^

error: use of deprecated macro `::modular_bitfield::BitfieldSpecifier`: use #[derive(Specifier)]. This alias will be removed in 0.13.
  --> tests/ui/derive_bitfield_specifier/deprecation_warning.rs:12:10
   |
12 | #[derive(BitfieldSpecifier)]
   |          ^^^^^^^^^^^^^^^^^

error: use of deprecated macro `::modular_bitfield::BitfieldSpecifier`: use #[derive(Specifier)]. This alias will be removed in 0.13.
  --> tests/ui/derive_bitfield_specifier/deprecation_warning.rs:18:10
   |
18 | #[derive(BitfieldSpecifier)]
   |          ^^^^^^^^^^^^^^^^^
